Added LC 1312 and updated the README.md

godmodevegeta opened the pull request.

Jiganesh reviewed (changes requested):
The problem is solved under dynamic programming folder.
Please add link to the problem at top of java solution file and put java file inside dynamic programming folder.
remove link added in readme file as it is already present in dynamic programming section

godmodevegeta commented:
Hi, based on your suggestions, I've
- Added link to the problem on top of the solution file
- Moved the file from string to dynamicProgramming
- Deleted the link from README.md

godmodevegeta commented on README.md:
```diff
@@ -574,7 +574,6 @@
 <details>
 <summary>Strings</summary>
 
```
I initially thought about adding a different problem I practised today. But then just did what you suggested.

Jiganesh commented:
Thank you !!
